x-pack/legacy/plugins/rollup/public/crud_app/sections/job_create/steps/step_metrics.js

```diff
@@ -250,28 +406,47 @@ export class StepMetricsUi extends Component {
           </EuiFlexItem>
         </EuiFlexGroup>
 
-        <EuiSpacer />
+        <EuiSpacer/>
 
         <FieldList
-          columns={this.listColumns}
+          columns={this.getListColumns()}
           fields={metrics}
           onRemoveField={this.onRemoveField}
-          emptyMessage={<p>No metrics fields added</p>}
-          addButton={(
-            <FieldChooser
-              buttonLabel={(
-                <FormattedMessage
-                  id="xpack.rollupJobs.create.stepMetrics.fieldsChooserLabel"
-                  defaultMessage="Add metrics fields"
+          emptyMessage={
+            <p>
+              {
+                i18n.translate(
+                  'xpack.rollupJobs.create.stepMetrics.emptyListLabel',
+                  { defaultMessage: 'No metrics fields added' },
+                )
+              }
+            </p>
+          }
+
+          addButton={
+            <EuiFlexGroup justifyContent={'flexStart'} alignItems={'center'}>
+              <EuiFlexItem grow={false}>
+                <FieldChooser
+                  key={'stepMetricsFieldChooser'}
```

Comment:
Minor nit: this can be a string `"stepMetricsFieldChooser"` without the enclosing braces. This comment applies to a few other places too.